qiskit/algorithms/evolvers/pvqd/pvqd.py

```diff
@@ -0,0 +1,364 @@
+
+    Example:
+
+        .. code-block:: python
+
+            import numpy as np
+
+            from qiskit import BasicAer
+            from qiskit.circuit.library import EfficientSU2
+            from qiskit.opflow import X, Z, I, MatrixExpectation
+
+            backend = BasicAer.get_backend("statevector_simulator")
+            expectation = MatrixExpectation()
+            hamiltonian = 0.1 * (Z ^ Z) + (I ^ X) + (X ^ I)
+            observable = Z ^ Z
+            ansatz = EfficientSU2(2, reps=1)
+            initial_parameters = np.zeros(ansatz.num_parameters)
+
+            time = 0.02
+            optimizer = L_BFGS_B()
+
+            # setup the algorithm
+            pvqd = PVQD(
+                ansatz,
+                initial_parameters,
+                timestep=0.01,
+                optimizer=optimizer,
+                quantum_instance=backend,
+                expectation=expectation
+            )
+
+            # specify the evolution problem
+            problem = EvolutionProblem(
+                hamiltonian, time, aux_operators=[hamiltonian, observable]
+            )
+
+            # and evolve!
+            result = pvqd.evolve(problem)
+
+    References:
+
+        [1] Stefano Barison, Filippo Vicentini, and Giuseppe Carleo (2021), An efficient
+            quantum algorithm for the time evolution of parameterized circuits,
+            `Quantum 5, 512 <https://quantum-journal.org/papers/q-2021-07-28-512/>`_.
+    """
+
+    def __init__(
+        self,
+        ansatz: Optional[QuantumCircuit] = None,
+        initial_parameters: Optional[np.ndarray] = None,
+        timestep: Optional[float] = None,
+        optimizer: Optional[Union[Optimizer, Minimizer]] = None,
+        expectation: Optional[ExpectationBase] = None,
+        initial_guess: Optional[np.ndarray] = None,
+        evolution: Optional[EvolutionSynthesis] = None,
+        use_parameter_shift: bool = True,
+        quantum_instance: Optional[Union[Backend, QuantumInstance]] = None,
+    ) -> None:
+        """
+        Args:
+            ansatz: A parameterized circuit preparing the variational ansatz to model the
+                time evolved quantum state.
+            initial_parameters: The initial parameters for the ansatz.
+            timestep: The time step.
+            optimizer: The classical optimizers used to minimize the overlap between
+                Trotterization and ansatz. Can be either a :class:`.Optimizer` or a callable
+                using the :class:`.Minimizer` protocol.
+            expectation: The expectation converter to evaluate expectation values.
+            initial_guess: The initial guess for the first VQE optimization. Afterwards the
+                previous iteration result is used as initial guess. If None, this is set to
+                a random vector with elements in the interval :math:`[-0.01, 0.01]`.
+            evolution: The evolution synthesis to use for the construction of the Trotter step.
```

Comment:
Should we mention the possible options here?

Reply:
I'll add a ref to the synthesis module 👍🏻